File: Leetcode/ARRAY/JAVA/Easy Problems/P_2373.java

public class P_2373 {

    public static int[][] largestLocal(int[][] grid) {
        int n = grid.length;
        int[][] maxLocal = new int[n - 2][n - 2];

        int row = 0, colom = 0;

        while (row < n - 2) {

            int max = 0;

            for (int i = row; i < row + 3; i++) {
                for (int j = colom; j < colom + 3; j++) {
                    if (max < grid[i][j]) {
                        max = grid[i][j];
                    }
                }
            }

            maxLocal[row][colom] = max;

            colom++;

            if (colom == n - 2) {
                colom = 0;
                row++;
            }

        }

        return maxLocal;

    }

    public static void main(String[] args) {
        int[][] grid = {{9, 9, 8, 1}, {5, 6, 2, 6}, {8, 2, 6, 4}, {6, 2, 2, 2}};
        int[][] result = largestLocal(grid);

        for (int i = 0; i < result.length; i++) {
            for (int j = 0; j < result[i].length; j++) {
                System.out.println(result[i][j]);
            }
            System.out.println();
        }

    }
}

File: Leetcode/ARRAY/JavaScript/Easy Problems/P_2373.js
var largestLocal = function (grid) {
    let n = grid.length;
    let rows = n - 2;
    let cols = n - 2;

    // Change to declaring the 2D array with a different method and initializing with zeros
    let maxLocal = Array.from({ length: rows }, () => new Array(cols).fill(0));

    let row = 0, colom = 0, temp = 0;

    while (row < n - 2) {
        let max = 0;
        for (let i = row; i < row + 3; i++) {
            for (let j = colom; j < colom + 3; j++) {
                if (max < grid[i][j]) {
                    max = grid[i][j];
                }
            }
        }
        maxLocal[row][colom] = max;

        colom++;
        temp++;

        if (colom == n - 2) {
            colom = 0;
            temp = 0;
            row++;
        }
    }

    return maxLocal;

};

let grid = [[1, 1, 1, 1, 1], [1, 1, 1, 1, 1], [1, 1, 2, 1, 1], [1, 1, 1, 1, 1], [1, 1, 1, 1, 1]];
console.log(largestLocal(grid));




